addons/external-dns/gcp/main.tf

```diff
@@ -0,0 +1,26 @@
+data "terraform_remote_state" "infra" {
+  backend = "local"
+  config = {
+    path = "../../../infra/gcp/terraform.tfstate.d/gcp-${var.cluster_id}-${var.gcp_k8s_regions[var.cluster_id]}/terraform.tfstate"
+  }
+}
+
+data "terraform_remote_state" "dns" {
+  backend = "local"
+  config = {
+    path = "credentials/terraform.tfstate"
```

Comment:
The credentials should not be part of `addons` - we should host all the dependencies to stand up the module under the given module and the actual "run" should only reference the module with the provided variables.

Reply:
See the reasonning on the previous comment. This can be moved to the generic modules folder, but I think it's cleaner to keep one generic external-dns module for the reasoning above.

Reply:
`module/gcp/external-dns ` comment